File: Beginner/lapinDrome/lapinDrome.java
import java.util.Scanner;
import java.util.Arrays;

public class lapinDrome{
	public static void main(String[] args){
		int T,L;
		String S,left,right;
		boolean ans;
		Scanner scan = new Scanner(System.in);

		while(true){
			T = scan.nextInt();

			if(T<=0 || T>100){
				System.out.println("T must be 1 <= T <= 100");
			}else{
				break;
			}
		}

		while(T>0){

			while(true){
				S = scan.next();

				if(S.length()<2 || S.length()>1000){
					System.out.println("String S must have length 2 <= S <= 1000");
				}else{
					break;
				}
			}

			L = S.length();

			if(L%2==0){
				left = S.substring(0,L/2);
				right = S.substring(L/2,L);
			}else{
				left = S.substring(0,L/2);
				right = S.substring((L/2)+1,L);
			}

			char temp[] = left.toCharArray();
			char temp2[] = right.toCharArray();
			Arrays.sort(temp);
			Arrays.sort(temp2);

			left = new String(temp);
			right = new String(temp2);

			if(left.compareTo(right)==0){
				System.out.println("YES");
			}else{
				System.out.println("NO");
			}

			T--;
		}
	}
}

File: Beginner/lapinDrome/lapinDrome.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

int main(void){
	int T,L;
	char S[1000];
	bool ans;

	while(1){
		scanf("%d",&T);

		if(T<=0 || T>100){
			printf("T must be 1 <= T <= 100\n");
		}else{
			break;
		}
	}

	while(T>0){

		while(1){
			scanf("%s",S);

			if(strlen(S)==1){
				printf("String S must be at least 2 characters in length\n");
			}else{
				break;
			}
		}

		L = strlen(S);
		int left[500] = {0};
		int right[500] = {0};

		for(int i=0;i<L;i++){
			if(i<L/2){
				left[S[i]-'a']++;
			}
			if(i>=((L+1)/2)){
				right[S[i]-'a']++;
			}
		}

		ans = true;

		for(int c=0;c<26;c++){
			if(left[c]!=right[c]){
				ans = false;
				break;
			}
		}

		if(ans==true){
			printf("YES\n");
		}else{
			printf("NO\n");
		}

		T--;
	}

	return 0;
}